hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/receiver/SchemaExistenceEnsurer.java

```diff
@@ -0,0 +1,51 @@
+package pl.allegro.tech.hermes.consumers.consumer.receiver;
+
+import java.time.Duration;
+import net.jodah.failsafe.Failsafe;
+import net.jodah.failsafe.RetryPolicy;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import pl.allegro.tech.hermes.api.Topic;
+import pl.allegro.tech.hermes.schema.SchemaException;
+import pl.allegro.tech.hermes.schema.SchemaRepository;
+import pl.allegro.tech.hermes.schema.SchemaVersion;
+
+
+public class SchemaExistenceEnsurer {
```

Comment:
Have you intentionally not used `SchemaOnlineChecksRateLimiter`?

Reply:
I've made the initial PR, it is not complete yet. Just wanted to see what is your opnion about removing completely `AvroMessageAnySchemaVersionContentWrapper.java` class. I will definitely use ratelimiter 👍